torch/csrc/autograd/VariableTypeUtils.h

```diff
@@ -111,18 +111,29 @@ inline Tensor as_view(const Tensor & base, Tensor tensor, bool is_differentiable
   auto base_var = Variable(base);
   if (base_var.is_view()) {
     // Set `view_func` using the root base as input.
-    // `view_func` is used to recover views in backward when as_strided is not supported.
+    // `view_func` is used to recover views in backward when either as_strided is not supported
+    // or the view function changes the dtype which is not recorded by as_strided
     // See Note [View + Inplace update on base tensor] and [View + Inplace update on view tensor]
     // for more details how we use this function in backward.
+    auto diff_view_meta = static_cast<DifferentiableViewMeta*>(torch::autograd::impl::get_autograd_meta(base_var));
     if (view_func.has_value()) {
       auto fn = view_func.value();
-      auto diff_view_meta = static_cast<DifferentiableViewMeta*>(torch::autograd::impl::get_autograd_meta(base_var));
       if (diff_view_meta->has_view_fn()) {
         auto prev_fn = diff_view_meta->view_fn();
-        view_func = [=](const at::Tensor& root_base) {
-          auto temp = prev_fn(root_base);
-          return fn(temp);
-        };
+        if (prev_fn == nullptr) {
+          view_func = [=](const at::Tensor& root_base) {
+            auto sizes = root_base.sizes().vec();
```

Comment:
Do you actually need to make copies with the `.vec()` here?

Reply:
no you're right updated